google-http-client/src/main/java/com/google/api/client/http/javanet/NetHttpRequest.java

```diff
@@ -111,4 +127,39 @@ public LowLevelHttpResponse execute() throws IOException {
       }
     }
   }
+
+  private void writeContentToOutputStream(OutputStream out)
+      throws IOException {
+    if (writeTimeout == 0) {
+      // If no timeout set, do a plain write
+      getStreamingContent().writeTo(out);
+    } else {
+      // Use and executor service and futures to handle this
+      final OutputStream outputStream = out;
+      final StreamingContent content = getStreamingContent();
+      final Callable<Boolean> writeContent = new Callable<Boolean>() {
+        @Override
+        public Boolean call() throws IOException {
+          content.writeTo(outputStream);
+          return Boolean.TRUE;
+        }
+      };
+      final ExecutorService executor = Executors.newSingleThreadExecutor();
+      final Future future = executor.submit(new FutureTask<Boolean>(writeContent), null);
```

Comment:
Does creating additional thread affect performance in a negative way?

Reply:
We'll have to investigate deeper. With the current implementation, we are only using the Future if we set a write timeout.